src/core/src/signature.rs

```diff
@@ -107,19 +110,33 @@ pub trait SigsTrait {
 
                 aa.windows(aa_ksize as usize).for_each(|n| {
                     let hash = crate::_hash_murmur(n, self.seed());
-                    self.add_hash(hash);
+                    seq_hashes.push(hash);
                 });
 
                 let rc_substr: Vec<u8> = rc.iter().cloned().skip(i).take(rc.len() - i).collect();
                 let aa_rc = to_aa(&rc_substr, hash_function.dayhoff(), hash_function.hp()).unwrap();
 
                 aa_rc.windows(aa_ksize as usize).for_each(|n| {
                     let hash = crate::_hash_murmur(n, self.seed());
-                    self.add_hash(hash);
+                    seq_hashes.push(hash);
                 });
             }
         }
 
+        Ok(seq_hashes)
+    }
+
+    fn add_sequence(&mut self, seq: &[u8], force: bool) -> Result<(), Error> {
+        let ready_hashes = match self.seq_to_hashes(seq, force) {
+            Ok(s) => s,
+            Err(e) => return Err(e),
+        };
```

Comment:
This block is the same implementation of the `?` operator, so you can do this instead:
```suggestion
        let ready_hashes = self.seq_to_hashes(seq, force)?;
```